### ISSUE TITLE
[explicit-function-return-type] allowExpressions should not allow methods

### ISSUE THREAD

**Issue by whatisaphone** [typescript-eslint/typescript-eslint#950]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/explicit-function-return-type": ["error", {"allowExpressions": true}]
  }
}
```

```TS
class C {
  public method = () => {};
}
const variable = () => {};
```

**Expected Result**

Both lines should get hit with the error: "Missing return type on function".

**Actual Result**

Only `variable` gets the error; `method` does not. `method` is erroneously allowed due to `allowExpressions`.

**Additional Info**

From reading old issues, it looks like the idea behind `allowExpressions` is to allow expressions, but not to allow pure declarations which consist of only a function expression ([source](https://github.com/typescript-eslint/typescript-eslint/issues/666#issuecomment-507463056)). This categorization is working great already for variable declarartions. I think the `allowExpressions` option should follow the same reasoning for both, and also treat a class property method declaration as a function declaration.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.1.0` |
| `@typescript-eslint/parser`        | `2.1.0` |
| `TypeScript`                       | `3.5.1` |
| `ESLint`                           | `6.3.0` |
| `node`                             | `8.10.0` |
| `npm`                              | `5.6.0` |
